pkg/argocd/argocd.go

```diff
@@ -481,19 +481,32 @@ func SetKustomizeImage(app *v1alpha1.Application, newImage *image.ContainerImage
 	return nil
 }
 
+// ImageIsAllowed checks whether img is declared in image-list annotation
+func ImageIsAllowed(img *image.ContainerImage, list *image.ContainerImageList) bool {
+	for _, i := range *list {
+		if i.ImageName == img.ImageName {
+			return true
+		}
+	}
+	return false
+}
+
 // GetImagesFromApplication returns the list of known images for the given application
 func GetImagesFromApplication(app *v1alpha1.Application) image.ContainerImageList {
```

Comment:
I am not really like such approach, because you change total sense of this function. I think it would be better to add validation [here](https://github.com/ziouf/argocd-image-updater/blob/14bdc8115a450d87812e4404952fa20431065057/pkg/argocd/update.go#L422). Either you need rename function and change comment